web/components/amount-input.tsx

```diff
@@ -34,46 +35,57 @@ export function AmountInput(props: {
     const isInvalid = !str || isNaN(amount)
     onChange(isInvalid ? undefined : amount)
   }
+
   const { width } = useWindowSize()
   const isMobile = (width ?? 0) < 768
   return (
-    <Col className={className}>
-      <label className="input-group mb-4">
-        <span className="bg-gray-200 text-sm">{label}</span>
-        <input
-          className={clsx(
-            'input input-bordered max-w-[200px] text-lg placeholder:text-gray-400',
-            error && 'input-error',
-            inputClassName
-          )}
-          ref={inputRef}
-          type="text"
-          pattern="[0-9]*"
-          inputMode="numeric"
-          placeholder="0"
-          maxLength={6}
-          autoFocus={!isMobile}
-          value={amount ?? ''}
-          disabled={disabled}
-          onChange={(e) => onAmountChange(e.target.value)}
-        />
-      </label>
+    <>
+      <Col className={className}>
+        <label>
+          <span
+            className={clsx(
+              'text-greyscale-4 absolute ml-2 mt-[9px]',
+              isMobile ? 'font-sm' : 'font-lg'
```

Comment:
In general, it's better to use css media queries to adjust based on screen size, e.g.  replace this line with`font-sm sm:font-lg`

It's faster, and it can be correct when server-side rendered even when we don't know the screen size yet, which could prevent a flicker.